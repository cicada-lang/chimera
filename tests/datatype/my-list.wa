List List::null
List List::cons(_head, tail) -- { List tail }

find _ {
  List List::null
  List List::cons(1, List::cons(2, List::cons(3, List::null)))
}

find [list, x, y, z] {
  list = List::cons(1, List::cons(2, List::cons(3, List::null)))
  list = List::cons(x, List::cons(y, List::cons(z, List::null)))
}

find [list, x, y, z] {
  list = List::cons(x, List::cons(y, List::cons(z, List::null)))
}
