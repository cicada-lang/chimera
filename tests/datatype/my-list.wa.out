[ _.0 ]
[ 
  [ 
    List::cons(1, List::cons(2, List::cons(3, List::null()))),
    1,
    2,
    3
  ]
]
[ 
  [ 
    List::cons(_.0, List::cons(_.1, List::cons(_.2, List::null()))),
    _.0,
    _.1,
    _.2
  ]
]
